chain/chain/src/stateless_validation/chunk_validation.rs

```diff
@@ -245,46 +246,54 @@ pub fn pre_validate_chunk_state_witness(
         )));
     }
 
-    // Verify that all proposed transactions are valid.
-    let new_transactions = &state_witness.new_transactions;
-    if !new_transactions.is_empty() {
```

Comment:
I don't think you need to have an additional check, you can just rely on `new_transactions` being empty when `RelaxedChunkValidation` is enabled

Reply:
I find it somewhat easier to reason about with the protocol feature thing. It does not require implicit knowledge that `new_transactions` are always empty past the feature stabilization.

Though on the other hand we probably should fail out chunks containing non-empty `new_transactions` right here :thinking: